vm/src/obj/objstr.rs

```diff
@@ -33,7 +34,36 @@ impl<T: ToString> From<T> for PyString {
     }
 }
 
-pub type PyStringRef = PyRef<PyString>;
+pub trait IntoPyStringRef {
+    fn into_pystringref(self, vm: &mut VirtualMachine) -> PyResult<PyStringRef>;
+}
+
+impl IntoPyStringRef for PyStringRef {
+    fn into_pystringref(self, _vm: &mut VirtualMachine) -> PyResult<PyStringRef> {
+        Ok(self)
+    }
+}
+
+impl IntoPyStringRef for String {
```

Comment:
These impls do seem a bit redundant since we already have similar ones for `PyString`. Could we do something like:

```rust
impl<T: Into<PyString>> IntoPyStringRef for T
```

Or to generalize it further:

```rust
impl<V: PyValue, T: Into<V>> IntoPyRef<V> for T
```

Reply:
I don't think it works.

Assuming you meant:

```
impl<T: Into<PyString>> TryIntoPyStringRef for T
{
    fn into_pystringref(self, vm: &mut VirtualMachine) -> PyResult<PyStringRef> {
        Ok(PyString::from(self).into_ref(vm))
    }
}
```

It complains because:

```
help: the trait `std::fmt::Display` is not implemented for `T`
```

And fixing that:

```
47 | impl TryIntoPyStringRef for PyStringRef {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `pyobject::PyRef<obj::objstr::PyString>`
```

if you allow it to use the version it derives then it ends up formatting the object using the default formatter and you get: ```"'str' object"``` instead of the string. (I was expected to lose object identity in that case, surprised to lose object value.)

Reply:
This default implementation for ToString is the issue:

```
impl<T> ToString for T where
    T: Display + ?Sized, 
```